protocol/messenger_handler.go

```diff
@@ -517,6 +517,22 @@ func (m *Messenger) HandleSyncInstallationContact(state *ReceivedMessageState, m
 		state.AllChats.Store(chat.ID, chat)
 	}
 
+	if contact.Added && !contact.HasAddedUs {
+		// This means we have an outgoing contact request that is pending.
+		// There might have been a response to this request already
+		// which wasn't backed up, so we're resending the contact request
+		// and hope for a response.
+
+		request := &requests.SendContactRequest{
+			ID:      types.Hex2Bytes(contact.ID),
+			Message: "Resending pending contact request",
```

Comment:
I guess this PR needs more discussion as per https://github.com/status-im/status-go/pull/3052#discussion_r1066295420

Put it on backlog now (because it's an edge case really), so I can focus on more urgent things